iam.tf

```
@@ -0,0 +1,64 @@
+########################################
+# I is for IAM, and this module creates two roles
+#
+# ECS Task Execution Role
+#   IAM role used by the task itself
+#    e.g. if your container wants to call other AWS services like S3, SQS, etc
+#    then those permissions would need to be covered by the TaskRole
+########################################
+data "aws_iam_policy_document" "assume" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "ecs_task" {
+  name_prefix        = local.ecs_task_iam_role_name_prefix
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+  tags               = var.tags
+}
+
+resource "aws_iam_role_policy" "additional" {
+  count = length(var.additional_ecs_task_policy_arns)
+
+  name_prefix = "${var.name}-ecs-task-policy-${count.index}"
```

```suggestion
  name_prefix = "${var.name}-ecs-task-policy-${count.index}-"
```
I'm not even sure we need `name_prefix` at all for this particular resource type but don't feel strongly